cirq/work/sampler.py

```diff
@@ -133,6 +134,55 @@ def sample(
 
         return pd.concat(results)
 
+    def sample_expectation_values(
+        self,
+        program: 'cirq.Circuit',
+        measurement_to_observables: Dict[
+            str, Union['cirq.PauliSumLike', List['cirq.PauliSumLike']]
+        ],
+        *,
+        num_samples: int,
+        params: 'cirq.Sweepable' = None,
+    ) -> List[List[float]]:
+        if num_samples <= 0:
+            raise ValueError(
+                f'Expectation values require at least one sample. Received: {num_samples}.'
+            )
+        samples = self.sample(program, repetitions=num_samples, params=params)
+        num_qubits = len(program.all_qubits())
+        num_param_values = len(list(study.to_resolvers(params)))
+        qmap = {q: i for i, q in enumerate(ops.QubitOrder.DEFAULT.order_for(program.all_qubits()))}
+
+        results = []
+        row_iter = samples.itertuples()
+        mto = {
+            key: [ops.PauliSum.wrap(o) for o in obs]
+            if isinstance(obs, List)
+            else [ops.PauliSum.wrap(obs)]
+            for key, obs in measurement_to_observables.items()
+        }
+        # Loop over sweeps
+        for _ in range(num_param_values):
+            state_vector_map = {}
+            for key in mto:
+                state_vector_map[key] = np.zeros(2 ** num_qubits, dtype=np.complex64)
+            # Aggregate results from all samples for the same parameter values
+            for __ in range(num_samples):
+                curr_row = next(row_iter)
+                for key in mto:
+                    state_vector_map[key][getattr(curr_row, key)] += 1
+            # Compute estimated expectation values from aggregated results
+            sweep_result = []
+            for key, observables in mto.items():
+                norm = np.linalg.norm(state_vector_map[key])
+                state_vector_map[key] /= norm
```

Comment:
Hmm...I don't understand this calculation. You are trying to construct a full statevector based on potentially partial measurements...

The problems that I see here is
1. nothing guarantees that the partial measurement of the selected qubits is going to add up to 1 probability 
2. even if we assume that the chosen subset of amplitudes fill up the whole state vector, we'd have to do an inverse Born rule kind of thing: 
```
# sum of observed frequencies for each bitstring
norm = state_vector_map[key].sum()
# relative probabilities 
state_vector_map[key] /= norm
# to amplitudes 
state_vector_map[key] = np.sqrt(state_vector_map[key])
```

However, I think we should go about this differently than reusing the `obs.expectation_from_state_vector` as we don't have a statevector/density matrix here. Is there a way to tie this together with the Observable Measurements functionality in cirq.work? Because that uses a Sampler and inserts the right type of rotations before measurements...

cc @mpharrigan

Reply:
You're right about the "inverse Born rule" step vs using the norm - I guess I missed this because all of my tests have uniform amplitudes. I'll commit a fix for it shortly.

The intent here is to construct a frequentist approximation of the state vector for the subset of qubits measured at each measurement event. Phase information is lost, but that's fine since we don't need it for estimating expectation values.

`state_vector_map[key]` may not be a "true" state vector, but it approximates one. Since `expectation_from_state_vector` does exactly what we want, I don't see why we wouldn't use it here unless there's an optimization possible for the pre-normalization state histogram.

Reply:
Relevant: #2623

Reply:
Added a new test with more imbalanced amplitudes and verified that it fails with 2-norm and passes with inverse Born rule.

Reply:
Here's my reasoning why this won't work - the key problem is that you lose out on relative phases for the amplitudes... 

```
import cirq
import numpy as np

a = cirq.NamedQubit("a")

alpha = 0.3 + 0.1j
beta = 1j * np.sqrt((1 - np.abs(alpha) ** 2))

# the true state vector 
psi = np.array([alpha, beta])
print(psi)
print(np.linalg.norm(psi))
print("Expectation value1: ", cirq.X(a).expectation_from_state_vector(psi, qubit_map={a: 0}))

# this is the resulting "state vector" from the "inverse born rule"
psi2 = np.array([np.abs(alpha), np.abs(beta)], dtype=np.complex64)

print(psi2)
print(np.linalg.norm(psi2))
print("Expectation value2: ", cirq.X(a).expectation_from_state_vector(psi2, qubit_map={a: 0}))

```